psst-gui/src/ui/public_user.rs

```diff
@@ -0,0 +1,368 @@
+        AppState, ArtistLink, Cached, Ctx, MixedView, Nav, PublicUser, WithCtx,
+    },
+    ui::{
+        artist, playlist, theme,
+        utils::{self, spinner_widget},
+    },
+    webapi::WebApi,
+    widget::{icons, Async, Empty, MyWidgetExt, RemoteImage},
+};
+use druid::{
+    im::Vector,
+    kurbo::Circle,
+    lens::Map,
+    widget::{Button, CrossAxisAlignment, Either, Flex, Label, List, Scroll},
+    Color, LensExt, Selector, UnitPoint, Widget, WidgetExt,
+};
+
+pub const LOAD_DETAIL: Selector<(PublicUser, AppState)> =
+    Selector::new("app.publicUser.load-detail");
+pub const FOLLOW_USER: Selector<Arc<str>> = Selector::new("app.publicUser.follow");
+pub const UNFOLLOW_USER: Selector<Arc<str>> = Selector::new("app.publicUser.unfollow");
+
+pub fn detail_widget() -> impl Widget<AppState> {
+    Async::new(spinner_widget, loaded_detail_widget, || Empty)
+        .lens(
+            Ctx::make(
+                AppState::common_ctx,
+                AppState::public_user_detail.then(PublicUserDetail::info),
+            )
+            .then(Ctx::in_promise()),
+        )
+        .on_command_async(
+            LOAD_DETAIL,
+            |d| WebApi::global().get_public_user_profile(d.0.id),
+            |_, data, q| data.public_user_detail.info.defer(q.0),
+            |_, data, r| {
+                data.public_user_detail
+                    .info
+                    .update((r.0 .0, r.1.map(|info| Cached::fresh(Arc::new(info)))))
+            },
+        )
+        .on_command_async(
+            FOLLOW_USER,
+            |username| WebApi::global().follow_user(&username),
+            |_, _, _| {},
+            |_, data, (_username, r)| {
+                if let Err(err) = r {
+                    data.error_alert(err);
+                } else {
+                    data.info_alert("User followed.");
+                    // Update the is_following state in the cached data
+                    if let Some(cached) = data.public_user_detail.info.resolved_mut() {
+                        let info = Arc::make_mut(&mut cached.data);
+                        info.is_following = Some(true);
+                        info.followers_count += 1;
+                    }
+                }
+            },
+        )
+        .on_command_async(
+            UNFOLLOW_USER,
+            |username| WebApi::global().unfollow_user(&username),
+            |_, _, _| {},
+            |_, data, (_username, r)| {
+                if let Err(err) = r {
+                    data.error_alert(err);
+                } else {
+                    data.info_alert("User unfollowed.");
+                    // Update the is_following state in the cached data
+                    if let Some(cached) = data.public_user_detail.info.resolved_mut() {
+                        let info = Arc::make_mut(&mut cached.data);
+                        info.is_following = Some(false);
+                        info.followers_count -= 1;
```

Comment:
I guess it makes sense that we update these values, but do we ever actually refresh these values elsewhere? Not that it matters, just curious. I'm trying to think about how Psst might cache things, but some fields shouldn't really need to be cached if they are dynamic, like follower counts, listener counter perhaps.